deployment/cloud-run/secure_api_server.py

```
@@ -479,34 +530,488 @@
+                    text_to_analyze = transcription_result.text if hasattr(transcription_result, 'text') else str(transcription_result)
+                finally:
+                    os.unlink(temp_path)
+
+        if not text_to_analyze:
+            api.abort(400, "Either text or audio file must be provided")
+
+        # Emotion Analysis
+        emotion_result = {}
+        try:
+            raw_emotion = predict_emotions(text_to_analyze, threshold=emotion_threshold)
+            emotion_result = normalize_emotion_results(raw_emotion)
```

Ruff (0.12.2)

Undefined name `normalize_emotion_results`

(F821)